docs/utils/threading/worker-pool.md

```diff
@@ -148,3 +148,217 @@ functionality in Thunder, which will be further described in a different documen
+
+ Error codes that can be returned:
+
+* `ERROR_NONE` - 0 (no errors)
+* `ERROR_UNKNOWN_KEY` - 22
+
+!!!warning
+    `ERROR_UNKNOWN_KEY` will be returned in case the job was not previously scheduled!
+
+```cpp
+uint32_t Revoke(const Core::ProxyType<IDispatch>& job, const uint32_t waitTime = Core::infinite) override
+        {
+            uint32_t result(_timer.Revoke(Timer(this, job)) ? Core::ERROR_NONE : Core::ERROR_UNKNOWN_KEY);
+
+            uint32_t report = _threadPool.Revoke(job, waitTime);
+
+            if (report == Core::ERROR_UNKNOWN_KEY) {
+                report = _external.Completed(job, waitTime);
+
+                if ( (report != Core::ERROR_UNKNOWN_KEY) && (result == Core::ERROR_UNKNOWN_KEY) ) {
+                    result = report;
+                }
+            }
+
+            return (result);
+        }
+```
+#### Example
+
+Below is test example of using it.
+```cpp
+void test_scheduler(Core::ProxyType<IDispatch> job) {
+    // Schedule the job to run 5 seconds from now
+    Core::Time scheduledTime = Core::Time::Now() + Core::TimeSpan::FromSeconds(5);
+    _workerpool.Schedule(scheduledTime, job);
+
+    // Wait for a while to see the scheduled job execution
+    Core::Thread::Sleep(Core::TimeSpan::FromSeconds(10));
+
+    // Reschedule the job to run 10 seconds from now
+    Core::Time rescheduledTime = Core::Time::Now() + Core::TimeSpan::FromSeconds(10);
+    _workerpool.Reschedule(rescheduledTime, job);
+
+    // Wait for a while to see the rescheduled job execution
+    Core::Thread::Sleep(Core::TimeSpan::FromSeconds(15));
+
+    // Revoke the job (cancel its execution)
+    _workerpool.Revoke(job);
+}
+```
+For more examples you can check `Thunder/Tests/unit/core/test_workerpool.cpp` file.
+
+
+### Scheduler class
+`Scheduler` is a small class designed to schedule jobs in the future. It has one public method `Schedule()`. It takes as arguments the `job` whose execution is to be scheduled in the future and the time after which the execution of the `job` should start.
+```cpp
+void Schedule(const Time& time, const ProxyType<IDispatch>& job) override {
+                _timer.Schedule(time, Timer(_pool, job));
+            }
+```
+The Scheduler uses the `Timer` class of the `Workerpool` class as a privileged class. How the `Timer` class is built and how it works can be seen in the `Timer.h` file.
+
+### Job class scheduling methods
+Each `job` has two methods by which we can reschedule them for future execution and cancel them.
+ These methods are:
+
+* `Reschedule()`
+* `Revoke()`
+
+#### Rescheduling job
+`Reschedule()` allows the scheduled `job` execution time to be changed. The id argument takes the time by which we should postpone the execution of the job. The method returns `true` if the operation is successful.
+```cpp
+bool Reschedule(const Core::Time& time)
```

Comment:
If we include a piece of code, I'd write a few sentences of what is going on in this code